### PR TITLE
chore: update proposal api

### PR DESCRIPTION
**Pull request by catngh**
**What does this PR do?**

- fix guild config dao proposal delete api

### CODE REVIEW

File: pkg/repo/dao_proposal/store.go [hunk 1]
```diff
@@ -44,3 +44,7 @@ func (pg *pg) UpdateDiscussionChannel(id int64, discussionChannelId string) erro
 func (pg *pg) DeleteById(id int64) error {
 	return pg.db.Delete(&model.DaoProposal{}, id).Error
 }
+
+func (pg *pg) BatchUpdateConfigIDByProposalID(id string) error {
```
Comment (trkhoi) [inline pkg/repo/dao_proposal/store.go:48]: no need

File: pkg/entities/guild_config_dao_proposal.go [hunk 2]
```diff
@@ -42,16 +49,20 @@ func (e *Entity) CreateProposalChannelConfig(req request.CreateProposalChannelCo
...
-			return nil, nil
-		}
 		e.log.Fields(logger.Fields{"guildId": guildId}).Error(err, "[entity.GetGuildConfigDaoProposalByGuildID] e.repo.GuildConfigDaoProposal.GetByGuildId failed")
 		return nil, err
 	}
 	return config, nil
 }
 
 func (e *Entity) DeleteGuildConfigDaoProposalByGuildID(req *request.DeleteGuildConfigDaoProposal) error {
+	// remove foreign key from proposals
```
Comment (trkhoi) [inline pkg/entities/guild_config_dao_proposal.go:59]: no need

File: pkg/repo/dao_proposal/pg.go [hunk 3]
```diff
@@ -9,4 +9,5 @@ type Store interface {
 	Create(model *model.DaoProposal) (*model.DaoProposal, error)
 	UpdateDiscussionChannel(id int64, discussionChannelId string) error
 	DeleteById(id int64) error
+	BatchUpdateConfigIDByProposalID(id string) error
```
Comment (trkhoi) [inline pkg/repo/dao_proposal/pg.go:12]: no need

File: migrations/schemas/20230111142608-alter_dao_proposal_nullable_config.sql [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+-- +migrate Up
+ALTER TABLE dao_proposal ALTER COLUMN guild_config_dao_proposal_id DROP NOT NULL;
```
Comment (trkhoi) [inline migrations/schemas/20230111142608-alter_dao_proposal_nullable_config.sql:3]: drop constraint foreign key in migration file

**Review by trkhoi — CHANGES_REQUESTED** [review]

**Review by trkhoi — APPROVED** [review]

**Review by hoangharry — APPROVED** [review]

**Pull request merged**